mypy/checker.py

```diff
@@ -1647,6 +1647,24 @@ def visit_class_def(self, defn: ClassDef) -> None:
                 with self.scope.push_class(defn.info):
                     self.accept(defn.defs)
             self.binder = old_binder
+            for base in typ.mro[1:]:
+                if base.name() != 'object' and base.defn.info:
+                    for method_name, method_symbol_node in base.defn.info.names.items():
+                        if method_name == '__init_subclass__':
+                            name_expr = NameExpr(defn.name)
+                            name_expr.node = base
+                            callee = MemberExpr(name_expr, '__init_subclass__')
+                            args = list(defn.keywords.values())
```

Comment:
At runtime, `metaclass` is _always_ removed from the arguments before passing to `__init_subclass__`. Could you please also add a test for this?